scripts/dependency-graph.js

```diff
@@ -11,6 +11,31 @@ const { run, CWD, pkg, fs, EOL } = require('./util.js')
 // npx -p @npmcli/stafftools gh repos --json | json -a name | sort > scripts/npm-cli-repos.txt
 const repos = readFileSync(join(CWD, 'scripts', 'npm-cli-repos.txt'), 'utf-8').trim().split(os.EOL)
 
+// Packages with known circular dependencies.  This is typically something with arborist as a dependency which is also in arborist's dev dependencies.  Not a problem if they're workspaces so we ignore repeats
+const circular = new Set(['@npmcli/mock-registry'])
```

Comment:
Should we just ignore devDependencies in the dependency graph, rather than opening the door to allowing circular dependencies?

Also it looks to me like https://github.com/npm/cli/blob/latest/DEPENDENCIES.md only lists dev dependencies for this repo. For example, I don't see `minipass-fetch-->npmcli-eslint-config["@npmcli/eslint-config"];` listed, despite https://github.com/npm/minipass-fetch/blob/e4df956af7c79cecc7dff8f3edb98cc263e0064c/package.json#L27.